src/component/Game.tsx

```diff
@@ -10,7 +10,22 @@ export const Game = () => {
   const [start, setStart] = useState<boolean>(false);
   const [incorrectLetters, setIncorrectLetters] = useState<number>(0);
 
-  const endGame = () => (currentWordIndex < words.length ? false : true);
+  const endGame = () => {
+    if (finalLetter()) {
+      setCurrentWordIndex(currentWordIndex + 1);
+      return true;
+    } else {
+      return currentWordIndex < words.length ? false : true;
+    }
+  };
+  const finalLetter = () => {
+    if (
+      currentWordIndex === words.length - 1 &&
+      currentInput === words[words.length - 1]
```

Comment:
These two lines of code don't ready very well; I can't tell what is happening here without trying harder than I want to.

Is there a way we could pull this check into a function so that we can gain some clarity as to what's happening via a good function name?

Reply:
That's kind of what I did though. In my first iteration I had this `if` statement inside of the `endGame` method, so I pulled it into its own method.

Reply:
Very fair, I was looking too closely at these two lines and lost content on where they were 👍